sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -763,11 +763,17 @@ object CatalogTable {
     }
   }
 
-  def readLargeTableProp(props: Map[String, String], key: String): Option[String] = {
+  def readLargeTableProp(
+      props: Map[String, String],
+      key: String,
+      failIfNumPartsMissing: Boolean = true): Option[String] = {
```

Comment:
when it's needed to be true? I think it's fine to always be more lenient, as the consequence is simply doing a schema inference, which is better than failing.